File: task1_compress.cpp
#include <iostream>
#include <string>
using namespace std;
 
string compress(const string& s) {
    string result = "";
    int i = 0;
    while (i < s.size()) {
        char c = s[i];
        int count = 0;
        while (i < s.size() && s[i] == c) {
            count++;
            i++;
        }
        result += c;
        result += to_string(count);
    }
    return result.size() < s.size() ? result : s;
}
 
int main() {
    string s;
    cout << "Введите строку: ";
    cin >> s;
    cout << "Результат: " << compress(s) << endl;
    return 0;
}

File: task1_compress.go
package main
 
import (
	"fmt"
	"strconv"
)
 
func compress(s string) string {
	result := ""
	i := 0
	for i < len(s) {
		c := s[i]
		count := 0
		for i < len(s) && s[i] == c {
			count++
			i++
		}
		result += string(c) + strconv.Itoa(count)
	}
	if len(result) < len(s) {
		return result
	}
	return s
}
 
func main() {
	var s string
	fmt.Print("Введите строку: ")
	fmt.Scan(&s)
	fmt.Println("Результат:", compress(s))
}
